### PR TITLE
fix: Include '.exe' suffix on windows

### PR DESCRIPTION
**Pull request by DLehenbauer**
Here's a first proposal for fixing #90 that uses the popular 'which' package to probe for the binary.

'which' handles both a binary at a specific path (e.g., when "viceDirectory" is set) as well as returning the first match from the user's path when just a binary name is given.

In both cases, 'which' will automatically match files with executable suffixes like ".exe" (based on 'env.PATHEXT').

(Using the Dockerfile environment, the same small handful of tests fail before/after this change.)

### CODE REVIEW

**Comment by DLehenbauer** [comment]
FYI - Looks like I may have regressed finding 'Mesen'.  Will take a look.  (Probably weekend?)

Couldn't find the emulator. Make sure your `cc65vice.viceDirectory` or `cc65vice.mesenDirectory` user setting is pointing to the directory containing emulator executables. /__w/vscode-cc65-debugger/vscode-cc65-debugger/src/__tests__/mesendir/Mesen.exe Error: not found: /__w/vscode-cc65-debugger/vscode-cc65-debugger/src/__tests__/mesendir/Mesen.exe

**Comment by empathicqubit** [comment]
Yeah, that does need to be fixed. Don't worry about the runahead tests at the moment. I may default that feature to off in the future since it might confuse people.

Are you able to see the test output in GitHub? I'm not sure about how the log permissions work.

**Comment by DLehenbauer** [comment]
I see the problem.  Being a Mono application, 'Mesen.exe' is technically not executable on Linux (i.e., lacks the '+x' permission that the 'which' package tests for.)

Bummer.

I'll find an alternative solution.

File: src/dbg/runtime.ts [hunk 1]
```diff
@@ -1269,6 +1269,38 @@ or define the location manually with the launch.json->mapFile setting`
         return this._getScope(this._currentPosition);
     }
 
+    private static async which(path: string): Promise<string> {
```
Comment (empathicqubit) [inline src/dbg/runtime.ts:1272]: Can you rename this to _which?

File: src/dbg/runtime.ts [hunk 2]
```diff
@@ -1269,6 +1269,38 @@ or define the location manually with the launch.json->mapFile setting`
...
+        // On Windows, the PATHEXT variable contains an ordered list of file extensions
+        // for executable (e.g., .exe, .com, .bat) that we need to additionally probe.
+        const pathExt = process.env.PATHEXT?.split(";") ?? [];
```
Comment (empathicqubit) [inline src/dbg/runtime.ts:1275]: Is there some other way to do this? Windows should already be doing this by default. I would feel more comfortable if this logic was coming from a library.

**Comment by DLehenbauer** [comment]
@empathicqubit - I hand-coded a slimmed down version of what the 'which' module does.  I believe it passes tests (except runahead, as mentioned above.)

**Comment by empathicqubit** [comment]
Maybe you could just do something different in the case of Mesen, and use which in the other cases?

**Comment by empathicqubit** [comment]
Try direct access -> try which -> hasbin

**Comment by empathicqubit** [comment]
Thanks for being the second contributor to the project!

**Comment by DLehenbauer** [comment]
@empathicqubit - It is my pleasure.  Thank you for making this awesome project!

**Pull request merged**